src/GitHub.VisualStudio/Commands/CreateGistCommand.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System;
+using System.ComponentModel.Composition;
+using System.Threading.Tasks;
+using GitHub.Commands;
+using GitHub.Logging;
+using GitHub.Services;
+using GitHub.Services.Vssdk.Commands;
+
+namespace GitHub.VisualStudio.Commands
+{
+    /// <summary>
+    /// Creates a gist from the currently selected text.
+    /// </summary>
+    [Export(typeof(ICreateGistCommand))]
+    public class CreateGistCommand : VsCommand, ICreateGistCommand
+    {
+        readonly Lazy<IDialogService> dialogService;
+        readonly Lazy<ISelectedTextProvider> selectedTextProvider;
+
+        [ImportingConstructor]
+        protected CreateGistCommand(IGitHubServiceProvider serviceProvider)
+            : base(CommandSet, CommandId)
+        {
+            dialogService = new Lazy<IDialogService>(() => serviceProvider.TryGetService<IDialogService>());
```

Comment:
I'm curious why we're not passing `Lazy<IDialogService>` (and others) into the importing constructor?

Reply:
Good question: basically this is how we were doing it before. I'll try using `Lazy<>`.